Epsilon/Services/LearningOutcomeCanvasResultService.cs

```diff
@@ -53,23 +61,18 @@ ILearningDomainService learningDomainService
         _learningDomainService = learningDomainService;
     }
 
-    public async IAsyncEnumerable<LearningDomainOutcomeResult> GetOutcomeResults(string studentId)
+    public async IAsyncEnumerable<LearningDomainSubmission> GetSubmissions(string studentId)
     {
         var submissionsTask = _canvasGraphQlApi.Query(Query, new Dictionary<string, object> { { "studentIds", studentId }, });
         var domainOutcomesTask = _learningDomainService.GetOutcomes();
 
         await Task.WhenAll(submissionsTask, domainOutcomesTask);
 
-        foreach (var submission in submissionsTask.Result.Courses.SelectMany(static c => c.Submissions.Nodes))
+        foreach (var submission in submissionsTask?.Result?.Courses?.SelectMany(static c => c.Submissions.Nodes))
```

Comment:
We should really do some null checks instead of adding these lazy ?'s Otherwise it will become a pain to debug in later versions, trust me on this one.

Reply:
True, but in the end it's a disadvantage of the architecture and  GraphQL. 
With in the same file it's determined what can be null and not. 

Those checks are only needed within the services that parse GraphQL data to our own models.

Reply:
For this instance I agree, but watch out for other values that CAN be null even in GraphQL response such as grade.